Packs/EmailCommunication/Scripts/PreprocessEmail/PreprocessEmail.py

```diff
@@ -238,45 +254,140 @@ def get_email_related_incident_id(email_related_incident_code, email_original_su
         incidents_details = get_incident_by_query(query)
         if len(incidents_details) == 0:
             code_is_unique = True
     return code
 
 
+def create_thread_context(email_code, email_cc, email_bcc, email_text, email_from, email_html,
+                          email_latest_message, email_received, email_replyto, email_subject, email_to,
+                          incident_id, attachments):
+    thread_number = str()
+    thread_found = False
+    try:
+        # Get current email threads from context if any are present
+        incident_context = demisto.executeCommand("getContext", {'id': incident_id})
+        incident_email_threads = dict_safe_get(incident_context[0], ['Contents', 'context', 'EmailThreads'])
+
+        # Check if there is already a thread for this email code
+        if incident_email_threads:
+            if isinstance(incident_email_threads, dict):
+                incident_email_threads = [incident_email_threads]
+
+            search_result = next((i for i, item in enumerate(incident_email_threads) if
+                                  item["EmailCommsThreadId"] == email_code), None)
+            if search_result is not None:
+                thread_number = incident_email_threads[search_result]['EmailCommsThreadNumber']
+                thread_found = True
+
+            if not thread_found:
+                # If no related thread is found, determine the highest thread number
+                max_thread_number = 0
+                for message in incident_email_threads:
+                    if int(message['EmailCommsThreadNumber']) > max_thread_number:
+                        max_thread_number = int(message['EmailCommsThreadNumber'])
+
+                thread_number = str(max_thread_number + 1)
+        else:
+            thread_number = '0'
+
+        if len(thread_number) == 0:
+            demisto.error('Failed to identify a Thread Number to set. Email not appended to incident context')
+
+        if attachments:
+            attachment_names = [attachment.get('name', '') for attachment in attachments]
+        else:
+            attachment_names = ["None"]
+
+        email_message = {
+            'EmailCommsThreadId': email_code,
+            'EmailCommsThreadNumber': thread_number,
+            'EmailCC': email_cc,
+            'EmailBCC': email_bcc,
+            'EmailBody': email_text,
+            'EmailFrom': email_from,
+            'EmailHTML': email_html,
+            'MessageID': email_latest_message,
+            'EmailReceived': email_received,
+            'EmailReplyTo': email_replyto,
+            'EmailSubject': email_subject,
+            'EmailTo': email_to,
+            'EmailAttachments': f'{attachment_names}',
+            'MessageDirection': 'inbound',
+            'MessageTime': get_utc_now().strftime("%Y-%m-%dT%H:%M:%SUTC")
+        }
+        # Add email message to context key
+        try:
+            demisto.executeCommand('executeCommandAt', {
```

Comment:
I was not able to find any other commands or scripts that would allow for setting context keys on a different incident other than the one being fetched and handled in pre-processing.  In this case a pre-existing incident is found that relates to the incoming email that's being processed, and the details of the new email need to be added as a context key on that related incident. `executeCommandAt` was the only way I could find to modify context on the related incident.

Please let me know if there's a more preferred way to do this that I missed.